bokehjs/src/lib/models/glyphs/webgl/regl_wrap.ts

```diff
@@ -400,3 +416,147 @@ function regl_marker(regl: any, marker_type: MarkerType): ReglRenderFunction {
     depth: {enable: false},
   })
 }
+
+
+function regl_rect_no_hatch(regl: any): ReglRenderFunction {
+  return regl({
+    vert: rect_vertex_shader,
+    frag: rect_fragment_shader,
+
+    attributes: {
+      a_position: {
+        buffer: regl.buffer([[-0.5, -0.5], [-0.5, 0.5], [0.5, 0.5], [0.5, -0.5]]),
+        divisor: 0,
+      },
+      a_center: {
+        buffer: regl.prop('center'),
+        divisor: 1,
+      },
+      a_width: (_: any, props: any) => {
+        return one_each_or_constant(props.width, 1, false, props.nmarkers)
+      },
+      a_height: (_: any, props: any) => {
+        return one_each_or_constant(props.height, 1, false, props.nmarkers)
+      },
+      a_angle: (_: any, props: any) => {
+        return one_each_or_constant(props.angle, 1, false, props.nmarkers)
+      },
+      a_linewidth: (_: any, props: any) => {
+        return one_each_or_constant(props.linewidth, 1, false, props.nmarkers)
+      },
+      a_line_color: (_: any, props: any) => {
+        return one_each_or_constant(props.line_color, 4, true, props.nmarkers)
+      },
+      a_fill_color: (_: any, props: any) => {
+        return one_each_or_constant(props.fill_color, 4, true, props.nmarkers)
+      },
+      a_line_join: (_: any, props: any) => {
```

Comment:
This is a lot of `any`s in one place (including `regl` argument).  This needs to be refined. Start with typing `regl` argument. Then you will have to provide type parameters to `regl<...>({...})` (see lines 93-101 in `node_modules/regl/regl.d.ts`).

Reply:
Removing the `any`s has been a big problem for me. Although my typescript is improving the ReGL typescript file is too advanced for my current understanding. Earlier today I decided that I need to take some time out to learn typescript properly.

Your link to the lines in the `regl.d.ts` file is a good starting point, and if I think about the problem in C++ templated class terms (which is what I am used to) I have an idea of how to proceed along the lines of declaring classes with which to instantiate the ReGL templated classes like `DrawConfig<stuff>`.

So I will try to make some progress on this, but don't be surprised if it is slow progress.

Reply:
I can't blame you as I have trouble myself trying to tie those types up, at least without sprinkling the code base with (otherwise) unnecessary type annotations. Unless you already made substantial progress, let's leave this for now and handle in a separate PR.

Reply:
Thanks, I'd prefer to leave it to a separate PR.  I intend to make it my next one.